src/Uno.UI.Composition/Composition/Uno/TransformableObjectExtensions.cs

```diff
@@ -52,7 +52,9 @@ public static Matrix4x4 GetTransform(this I3DTransformableObject transformableOb
 		var rotation = transformableObject.RotationAngle;
 		if (rotation is not 0)
 		{
+			transform *= Matrix4x4.CreateTranslation(-transformableObject.CenterPoint);
 			transform *= Matrix4x4.CreateFromAxisAngle(transformableObject.RotationAxis, rotation);
+			transform *= Matrix4x4.CreateTranslation(transformableObject.CenterPoint);
 		}
```

Comment:
@ramezgerges were you able to create a visual test for this particular behavior?

Reply:
@jeromelaban Pretty sure you meant to tag me. I will create an issue to track writing a test. This PR was to quickly unblock RefreshContainer.

Reply:
Ah yes, it was indeed for you :) Thanks!